routers/repo/migrate.go

```diff
@@ -171,3 +174,25 @@ func MigratePost(ctx *context.Context, form auth.MigrateRepoForm) {
 
 	handleMigrateError(ctx, ctxUser, err, "MigratePost", tplMigrate, &form)
 }
+
+// CancelMigration cancel a running migration
+func CancelMigration(ctx *context.Context) {
+	if !ctx.Repo.IsOwner() {
+		ctx.Error(http.StatusNotFound)
+		return
+	}
+
+	if ctx.Repo.Repository.Status != models.RepositoryBeingMigrated {
+		ctx.Error(http.StatusConflict, "repo already migrated")
+		return
+	}
+
```

Comment:
lunny: "We should send a stop singal to the migration go routine if it's migrating."

Reply:
from discord: ..."Basically we'd need a publish/subscribe system (in redis terms)."...

Reply:
@lunny since people are going to use "delete repo" anyway after asking in discord & co. - should we add this now and leave a new issue open to refactor the tasks to be able cancel on task level?